gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/predicates/DatasetHiveSchemaContainsNonOptionalUnion.java

```diff
@@ -42,16 +42,19 @@
 public class DatasetHiveSchemaContainsNonOptionalUnion<T extends Dataset> implements CheckedExceptionPredicate<T, IOException> {
   private final HiveRegister hiveRegister;
   private final Pattern pattern;
+  private final String optionalDbName;
 
   public static final String PREFIX = DatasetHiveSchemaContainsNonOptionalUnion.class.getName();
   /**
    * 1st match group is assumed to be the DB and the 2nd match group the Table for the pattern
    */
   public static final String PATTERN = PREFIX + ".db.table.pattern";
+  public static final String OPTIONAL_DB_NAME = PREFIX + ".db.optionalDbName";
 
   public DatasetHiveSchemaContainsNonOptionalUnion(Properties properties) {
     this.hiveRegister = getHiveRegister(new State(properties));
     this.pattern = Pattern.compile(properties.getProperty(PATTERN));
+    this.optionalDbName = properties.getProperty(OPTIONAL_DB_NAME, null);
```

Comment:
Instead of fetching and handling the default with null, can we use the java class Optional instead? Similar to the Optional<HiveTable>  below in the getTable.

Reply:
Updated to Optional